src/optimizer/query_to_operator_transformer.cpp

```diff
@@ -97,14 +98,20 @@ void QueryToOperatorTransformer::Visit(parser::SelectStatement *op, parser::Pars
         output_expr_ = std::move(filter_expr);
       }
     }
-  }
+  } else if (op->IsSelectDistinct()) {
+    // SELECT DISTINCT a1 FROM A should be transformed to
+    // SELECT a1 FROM A GROUP BY a1
+    // Assumption: SELECT DISTINCT on columns (not expressions)
```

Comment:
What does it take to not making the assumption? e.g. `SELECT DISTINCT(x*y) FROM table` might sometimes be a valid use case.

Reply:
I think this assumption might be out of date. `x*y` should be captured by the select column. Rewriting `SELECT DISTINCT(x*y) FROM table` can be rewritten to `SELECT x*y FROM table GROUP BY (x*y)`.